codes/classical/q-ary_digits/reed_solomon.yml

```diff
@@ -22,23 +22,43 @@ features:
+  - 'Although using iFFT has its counterpart iNNT for finite fields, the decoding is usually standard polynomial interpolation in \(k=\mathcal{O}(n\log^2 (n))\). However, in erasure decoding, encoded values are only erased in \(r\) points, which is a specific case of polynomial interpolation and can be done in \(\mathcal{O}(n\log (n)\) by computing product of received polynomial and erasure locator polynomial and using long division to find original polynomial. Long division step can be omitted to increase speed further by only dividing the derivative of the product polynomial, and derivative of erasure locator polynomial evaluated at erasure locations.'
+  - 'McEliece Public Key Cryptosystem \cite{manual:{McEliece, R.J.: A public-key cryptosystem based on algebraic coding theory. DSN Progress Report pp. 114–116 (1978).}} uses Algebraic Coding Theory to secure communications against eavesdropping attack, in which private keys are generator matrices of linear codes, i.e., \(G\). Public Keys shared to outside world are scrambled and permutated versions of \(G\), i.e., \(G^\prime=SGP\). Data to be encrypted, \(\bm{u}\), is multiplied by public key and added intentional errors \(\bm{e}\), i.e., \(x=\bm{u}G^\prime+\bm{e}\). Upon receiving encrypted data, private key owner can apply inverse permutation \(P^{-1}\) to \(\bm{x}\), decode the scrambled message given the presence of \(e\) errors and finally unscramble to obtain \(\bm{u}\). Security parameters of the system are \(n\) and \(e\), hence MDS codes, such as generalized Reed-Solomon codes may provide same security level for shorter key sizes and private key owner can decode arguably fast enough using known decoding algorithms.'
+  - 'Initial construction of McEliece Public Key Cryptosystem was based on Goppa codes which are subfield subcode of GRS codes. Public Key Cryptosystem designs based on Generalized Reed-Solomon codes first proposed by Niederreiter \cite{manual:{H. Niederreiter. Knapsack-type cryptosystems and algebraic coding theory. Prob- lems Control Inform. Theory, 15(2):159–166, 1986.}} which replaced generator matrix with parity check matrix, were proven to be insecure \cite{manual:{V.M. Sidelnikov and S.O. Shestakov. On the insecurity of cryptosystems based on generalized Reed-Solomon codes. Discrete Math. Appl., 1(4):439–444, 1992.}} since public key exposes algebraic structure of Reed-Solomon code. More recent works focuses on methods to mask the algebraic structure using subcodes of generalized Reed-Solomon codes \cite{manual:{T. P. Berger and P. Loidreau, “How to Mask the Structure of Codes for a Cryptographic Use,” Designs, Codes and Cryptography, vol. 35, pp. 63–79, 2005.}}. Lately a key-recovery attack was developed in \cite{arXiv:1307.6458} for a variant of masking method proposed by \cite{arXiv:1108.2462}.'
 
 relations:
   parents:
     - code_id: generalized_reed_solomon
+    - code_id: codes_for_distributed_storage
```

Comment:
distributed_storage What is the highest RS code family that can have this as parent? Can put there.